sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -2107,13 +2120,14 @@ private void SetNameFieldsIfNull()
                                 (IDownloadedContent)response.Value,
                                 response.GetRawResponse());
                         },
-                        (ETag? eTag) => new BlobRequestConditions { IfMatch = eTag },
                         async (bool async, CancellationToken cancellationToken)
                             => await GetPropertiesInternal(conditions: default, async, cancellationToken).ConfigureAwait(false),
+                        allowModifications,
+                        expectedETag,
                         blobProperties.Value.ContentLength,
                         position,
                         bufferSize,
-                        conditions);
+                        readConditions);
```

Comment:
What happens if someone's using a LeaseId?  Couldn't that be a breaking change because they read until the lease expires/is broken which doesn't change the ETag?

Reply:
LeaseId isn't applicable to this operation... (another reason to do something about BlobRequestConditions we discussed the other day).
See https://docs.microsoft.com/en-us/rest/api/storageservices/lease-blob#remarks
It applies to updates/deletes but not reads.

Reply:
It's not required, but it can be used.  See from that same link:

> It's not necessary to include the lease ID for GET operations on a blob that has an active lease. However, all GET operations support a conditional lease parameter, where the operation only proceeds if the lease ID included with the request is valid.

Reply:
Makes sense. I'll add test coverage for lease cases for both blobs and shares.

Reply:
After giving it second thought... Wouldn't it be awkward if we were able to open the stream (i.e. initial get properties succeeded) but in the middle of reading the stream lease got invalid?
Similar concern applies to other If-* conditions, e.g. based on LMT. The opening of the stream might satisfy them, but subsequent reads might fall outside of temporal condition range and fail. 
I think all these should be applied to initial action - the get properties. Afterwards it should give back bytes as long as it can read them consistently. So that the semantics is close to single shot get blob.

Reply:
I don't think so - especially for leases.  Imagine the scenario where I take a lease to ensure I've got sole access to a 10gb file.  I'd want to stop processing immediately if I lose the lease so I can go reacquire or at least stop wasting my time on a file I no longer have exclusive control over.

This the kind of thing I worry might present difficult semantic breaking changes.